File: cpp/abc086_c.cpp
#include<iostream>
#include<vector>
#include<cstdlib> // absを使うため
using namespace std;

int main()
{
  int N; cin >> N;
  vector<int> t(N+1), x(N+1), y(N+1);

  t[0] = x[0] = y[0] = 0; // 初期状態
  for(int i=1; i<=N; i++) cin >> t[i] >> x[i] >> y[i];

  bool can_travel = true;
  for(int i=0; i<N; i++) {
    int dt = t[i+1] - t[i];
    // 絶対値はabsで取得可能
    int dist = abs(x[i+1] - x[i])+ abs(y[i+1] -y[i]);

    if(dist > dt) can_travel = false;
    if ((x[i + 1] + y[i + 1]) % 2 == (x[i] + y[i]) % 2) {
      // 偶数が同じ場合はdtが偶数でなければならない
      if (dt % 2 != 0) can_travel = false;
    } else {
      // 偶奇が異なる場合はdtが奇数でなければならない
      if (dt % 2 != 1) can_travel = false;
    }
  }
  if(can_travel) cout << "Yes" << endl;
  else cout << "No";
  return 0;
}

File: kt/abc086_c.kt
fun main() {
  val n = readLine()!!.toInt()
  val ary = Array(n + 1) { IntArray(3) { 0 }}
  repeat(n) {
    val (t, x, y) = readLine()!!.split(" ").map { it.toInt() }
    ary[it][0] = t
    ary[it][1] = x
    ary[it][2] = y
  }
  val result = ary.sortedBy { it[0] }
  for (i in 0 until n) {
    val time = result[i + 1][0] - result[i][0]
    val dist = Math.abs((result[i + 1][1] + result[i + 1][2]) - (result[i][1] + result[i][2]))
    if (dist > time || dist % 2 != time % 2) {
      println("No")
      return
    }
  }
  println("Yes")
}